### PR TITLE
add GADM layers

### PR DESCRIPTION
**Pull request by solomon-negusse**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Make sure you are requesting to pull a topic/feature/bugfix branch (right side). Don't request your master!
- [ ] Make sure you are making a pull request against the develop branch (left side). Also you should start your branch off our develop.
- [ ] Check the commit's or even all commits' message styles matches our requested structure.
- [ ] Check your code additions will fail neither code linting checks nor unit test.

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

-
-
-

## Does this introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: src/main/scala/org/globalforestwatch/layers/GADMadm0.scala [hunk 1]
```diff
@@ -3,16 +3,264 @@ package org.globalforestwatch.layers
 import org.globalforestwatch.grids.GridTile
 
 case class GADMadm0(gridTile: GridTile, kwargs: Map[String, Any])
```
Comment (jterry64) [inline src/main/scala/org/globalforestwatch/layers/GADMadm0.scala:5]: It's not part of this PR per se, but for some reason I feel something is off about the casing... should ie be like GADMAdm0? GadmAdm0? It's hard with acronyms.

File: src/main/scala/org/globalforestwatch/summarystats/afi/AFiAnalysis.scala [hunk 2]
```diff
@@ -42,17 +41,36 @@ object AFiAnalysis extends SummaryAnalysis {
...
-
     // TODO somehow convert AFiSummary to AFiData
     import spark.implicits._
 
-    val summaryDF = AFiDF.getFeatureDataFrame(summaryRDD, spark)
+    val summaryDF = AFiDF
+      .getFeatureDataFrame(summaryRDD, spark)
+      .withColumn("gadm_id", when(col("location_id") =!= -1, lit("")).otherwise(col("gadm_id")))
+
+    val gadmAgg = summaryDF
```
Comment (jterry64) [inline src/main/scala/org/globalforestwatch/summarystats/afi/AFiAnalysis.scala:51]: maybe I'm reading, but I think you may have gotten it reversed - for GADM we want to group by GADM id

**Review by jterry64 — CHANGES_REQUESTED** [review]
It'd be good to see a sample output to make sure this is the right requirements

**Comment by solomon-negusse** [comment]
@jterry64: Addressed PR comments and points we discussed during pairing.

File: src/main/scala/org/globalforestwatch/summarystats/afi/AFiAnalysis.scala [hunk 3]
```diff
@@ -42,17 +41,48 @@ object AFiAnalysis extends SummaryAnalysis {
...
 //    }.unify
 
-
     // TODO somehow convert AFiSummary to AFiData
     import spark.implicits._
 
-    val summaryDF = AFiDF.getFeatureDataFrame(summaryRDD, spark)
+    val summaryDF = AFiDF
+      .getFeatureDataFrame(summaryRDD, spark)
+      .withColumn(
```
Comment (jterry64) [inline src/main/scala/org/globalforestwatch/summarystats/afi/AFiAnalysis.scala:49]: What case here are you dealing with by checking if null is in the GADM ID?

File: src/main/scala/org/globalforestwatch/summarystats/afi/AFiAnalysis.scala [hunk 4]
```diff
@@ -42,17 +41,48 @@ object AFiAnalysis extends SummaryAnalysis {
...
+        sum("total_area").alias("total_area"),
+        max("status_code").alias("status_code"),
+        concat_ws(", ", collect_list(when(col("location_error").isNotNull && col("location_error") =!= "", col("location_error")))).alias("location_error")
+      )
+
+
+    val gadmAgg = summaryDF
+      .filter($"location_id" === -1)
+      .groupBy($"list_id")
+      .agg(
```
Comment (jterry64) [inline src/main/scala/org/globalforestwatch/summarystats/afi/AFiAnalysis.scala:66]: Would it be possible to move this agg to a function we can apply to both DFs? I think they're the same. That might make it easier to read.

**Review by jterry64 — APPROVED** [review]
Looks good! Just some minor comments and questions.

File: src/main/scala/org/globalforestwatch/summarystats/afi/AFiAnalysis.scala (hunk 4 above)
Comment (solomon-negusse) [reply src/main/scala/org/globalforestwatch/summarystats/afi/AFiAnalysis.scala]: Good call, done.

File: src/main/scala/org/globalforestwatch/summarystats/afi/AFiAnalysis.scala (hunk 3 above)
Comment (solomon-negusse) [reply src/main/scala/org/globalforestwatch/summarystats/afi/AFiAnalysis.scala]: I'm doing that to consistently set the column to null (specifically doing that to merge scenarios of <ABC.null.null> and <ABC.123.null>  I encountered in an example run).
